runtime/compiler/optimizer/abstractinterpreter/AbsInterpreter.cpp

```diff
@@ -0,0 +1,3072 @@
+   }
+
+TR::AbsValue* TR::AbsBlockInterpreter::createStringObject(TR::SymbolReference* symRef, TR_YesNoMaybe isNonNull)
+   {
+   TR::VPClassPresence *classPresence = isNonNull == TR_yes ? TR::VPNonNullObject::create(vp()) : NULL;
+   TR::VPClassType *stringType = symRef ? TR::VPConstString::create(vp(), symRef) : NULL;
+
+   return new (region()) TR::AbsVPValue(vp(), TR::VPClass::create(vp(), stringType, classPresence, NULL, NULL, NULL), TR::Address);
+   }
+
+TR::AbsValue* TR::AbsBlockInterpreter::createIntConst(int32_t value)
+   {
+   return new (region()) TR::AbsVPValue(vp(), TR::VPIntConst::create(vp(), value), TR::Int32);
+   }
+
+TR::AbsValue* TR::AbsBlockInterpreter::createLongConst(int64_t value)
+   {
+   return new (region()) TR::AbsVPValue(vp(), TR::VPLongConst::create(vp(), value), TR::Int64);
+   }
+
+TR::AbsValue* TR::AbsBlockInterpreter::createIntRange(int32_t low, int32_t high)
+   {
+   return new (region()) TR::AbsVPValue(vp(), TR::VPIntRange::create(vp(), low, high), TR::Int32);
+   }
+
+TR::AbsValue* TR::AbsBlockInterpreter::createLongRange(int64_t low, int64_t high)
+   {
+   return new (region()) TR::AbsVPValue(vp(), TR::VPLongRange::create(vp(), low, high), TR::Int64);
+   }
+
+TR::AbsValue* TR::AbsBlockInterpreter::createTopInt()
+   {
+   return new (region()) TR::AbsVPValue(vp(), TR::VPIntRange::create(vp(), INT32_MIN, INT32_MAX), TR::Int32);
+   }
+
+TR::AbsValue* TR::AbsBlockInterpreter::createTopLong()
+   {
+   return new (region()) TR::AbsVPValue(vp(), TR::VPLongRange::create(vp(), INT64_MIN, INT64_MAX), TR::Int64);
+   }
+
+TR::AbsValue* TR::AbsBlockInterpreter::createTopDouble()
+   {
+   return new (region()) TR::AbsVPValue(vp(), NULL, TR::Double);
+   }
+
+TR::AbsValue* TR::AbsBlockInterpreter::createTopFloat()
+   {
+   return new (region()) TR::AbsVPValue(vp(), NULL, TR::Float);
+   }
+
+TR::AbsValue* TR::AbsBlockInterpreter::createTopObject()
+   {
+   return createObject(comp()->getObjectClassPointer(), TR_maybe);
+   }
+
+bool TR::AbsBlockInterpreter::isNullObject(TR::AbsValue* v)
+   {
+   TR::AbsVPValue* value = static_cast<TR::AbsVPValue*>(v);
+   return value->getConstraint() && value->getConstraint()->isNullObject();
+   }
+
+bool TR::AbsBlockInterpreter::isNonNullObject(TR::AbsValue* v)
+   {
+   TR::AbsVPValue* value = static_cast<TR::AbsVPValue*>(v);
+   return value->getConstraint() && value->getConstraint()->isNonNullObject();
+   }
+
+bool TR::AbsBlockInterpreter::isArrayObject(TR::AbsValue* v)
+   {
+   TR::AbsVPValue* value = static_cast<TR::AbsVPValue*>(v);
+   return value->getConstraint() && value->getConstraint()->asClass() && value->getConstraint()->getArrayInfo();
+   }
+
+bool TR::AbsBlockInterpreter::isObject(TR::AbsValue* v)
+   {
+   TR::AbsVPValue* value = static_cast<TR::AbsVPValue*>(v);
+   return value->getConstraint() && value->getConstraint()->asClass();
```

Comment:
Changed both conditions in `instanceof()` and `checkcast()` to just check if there is a constraint, and removed `isObject()` and `isNonNullObject()` since they are not used anymore.

I guess the former contributor misunderstood null and nonnull constraint and thought the reference should be either one of them. However, I checked value propagation doc and the null and nonnull constraint just indicate we are certain the reference is null or nonnull, and there is one more situation that we are not sure about the nullness, so I removed `isNonNullObject()` from the conditions of both callers.